### PR TITLE
Add dynamic exclude product in admin page

### PR DESCRIPTION
**Pull request by mdavid1297**
Add dynamic exclude product attributes and options in admin page
Add new Menu(Extended Search) with 2 SubMenus

### CODE REVIEW

File: src/Resources/config/grid/grid_exclude_attribute.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+sylius_grid:
+    grids:
+        cr_sylius_extended_elasticsearch_plugin_admin_exclude_attribute_index:
+            driver:
+                name: doctrine/orm
+                options:
+                    class: App\Entity\Product\ProductAttribute
+            fields:
+#                id:
```
Comment (jbcr) [inline src/Resources/config/grid/grid_exclude_attribute.yaml]: To delete

File: src/Resources/config/grid/grid_exclude_attribute.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                type:
+                    type: string
+                    label: sylius.ui.type
+                filterExcluded:
+                    type: twig
+                    label: bitbag_sylius_elasticsearch_plugin.ui.filter_excluded
+                    options:
+                        template: "@SyliusUi/Grid/Field/yesNo.html.twig"
+                    
+#                configuration:
```
Comment (jbcr) [inline src/Resources/config/grid/grid_exclude_attribute.yaml]: To delete

**Review by jbcr — CHANGES_REQUESTED** [review]

File: src/Resources/config/grid/grid_exclude_attribute.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+sylius_grid:
+    grids:
+        cr_sylius_extended_elasticsearch_plugin_admin_exclude_attribute_index:
+            driver:
+                name: doctrine/orm
+                options:
+                    class: App\Entity\Product\ProductAttribute
```
Comment (jbcr) [inline src/Resources/config/grid/grid_exclude_attribute.yaml]: Replace this value with the Sylius product attribute property

File: src/Resources/config/grid/grid_exclude_option.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+sylius_grid:
+    grids:
+        cr_sylius_extended_elasticsearch_plugin_admin_exclude_option_index:
+            driver:
+                name: doctrine/orm
+                options:
+                    class: App\Entity\Product\ProductOption
```
Comment (jbcr) [inline src/Resources/config/grid/grid_exclude_option.yaml]: Replace this value with the Sylius product option property

**Review by jbcr — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -123,6 +123,21 @@ parameters:
 By default all options and attributes are indexed. After you change these parameters, remember to run `bin/console fo:el:po` command again
 (a shortcut for `fos:elastica:populate`).
 
+### Exclude attributes and options dynamically from the search filter
```
Comment (jbcr) [inline README.md]:
```suggestion
### Exclude attributes and options dynamically from the search filter

To allow dynamic exclusion of product attributes (or product options), the `App\Entity\Product\ProductAttribute` class must be implement the `CodeRhapsodie\SyliusExtendedElasticsearchPlugin\Finder\FinderExcludable` interface

For example:

```php
<?php
//...
use CodeRhapsodie\SyliusExtendedElasticsearchPlugin\Finder\FinderExcludable

class ProductAttribute implements FinderExcludable
{
//...
}
```

If it's not the case(`ProductAttribute` doesn't implement `FinderExcludable`), the product attribute exclusion menu list will not be visible and will not appear.

It works the same way to allow the dynamic exclusion of `ProductOption`.
```

File: src/Controller/ProductAttributeController.php [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    private $repository;
+
+    public function __construct(RepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    /**
+     * @Route("/admin/search/excluded-attributes/toggle/{id}", name="cr_sylius_extended_elasticsearch_plugin_admin_exclude_attribute_toggle")
```
Comment (jbcr) [inline src/Controller/ProductAttributeController.php]: To delete, the routing configuration is in YAML file.

File: src/Controller/ProductOptionController.php [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+{
+    private $repository;
+
+    public function __construct(RepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    /**
+     * @Route("/admin/search/excluded-options/toggle/{id}", name="cr_sylius_extended_elasticsearch_plugin_admin_exclude_option_toggle")
```
Comment (jbcr) [inline src/Controller/ProductOptionController.php]: To delete

File: src/Resources/config/routing.yml [hunk 8]
```diff
@@ -18,3 +18,57 @@ bitbag_sylius_elasticsearch_plugin_shop_search:
     defaults:
         _controller: bitbag_sylius_elasticsearch_plugin.controller.action.shop.search
         template: "@CodeRhapsodieSyliusExtendedElasticsearchPlugin/Shop/search.html.twig"
+
+cr_sylius_extended_elasticsearch_plugin_admin_exclude_attribute_index:
+    path: /admin/search/exclude-attributes
```
Comment (jbcr) [inline src/Resources/config/routing.yml]:
```suggestion
    path: '/%sylius_admin.path_name%/search/exclude-attributes'
```

File: src/Resources/config/routing.yml [hunk 9]
```diff
@@ -18,3 +18,57 @@ bitbag_sylius_elasticsearch_plugin_shop_search:
...
+            grid: cr_sylius_extended_elasticsearch_plugin_admin_exclude_attribute_index
+            vars:
+                all:
+                    subheader: sylius.ui.manage_attributes_of_your_products
+                index:
+                    icon: 'cubes'
+
+
+cr_sylius_extended_elasticsearch_plugin_admin_exclude_attribute_toggle:
+    path: /admin/search/excluded-attributes/toggle/{id}
```
Comment (jbcr) [inline src/Resources/config/routing.yml]:
```suggestion
    path: '/%sylius_admin.path_name%/search/excluded-attributes/toggle/{id}'
```

File: src/Resources/config/routing.yml [hunk 10]
```diff
@@ -18,3 +18,57 @@ bitbag_sylius_elasticsearch_plugin_shop_search:
...
+    controller: CodeRhapsodie\SyliusExtendedElasticsearchPlugin\Controller\ProductAttributeController::modifyFilterExcluded
+    defaults:
+        _sylius:
+            section: admin
+            permission: true
+    requirements:
+        id: \d+
+
+cr_sylius_extended_elasticsearch_plugin_admin_exclude_option_index:
+    path: /admin/search/exclude-options
```
Comment (jbcr) [inline src/Resources/config/routing.yml]:
```suggestion
    path: '/%sylius_admin.path_name%/search/exclude-options'
```

File: src/Resources/config/routing.yml [hunk 11]
```diff
@@ -18,3 +18,57 @@ bitbag_sylius_elasticsearch_plugin_shop_search:
...
+            redirect: update
+            grid: cr_sylius_extended_elasticsearch_plugin_admin_exclude_option_index
+            vars:
+                all:
+                    subheader: sylius.ui.manage_configuration_options_of_your_products
+                index:
+                    icon: 'options'
+
+cr_sylius_extended_elasticsearch_plugin_admin_exclude_option_toggle:
+    path: /admin/search/excluded-options/toggle/{id}
```
Comment (jbcr) [inline src/Resources/config/routing.yml]:
```suggestion
    path: '/%sylius_admin.path_name%/search/excluded-options/toggle/{id}'
```

**Review by jbcr — CHANGES_REQUESTED** [review]

File: README.md [hunk 12]
```diff
@@ -123,6 +123,30 @@ parameters:
...
+    //...
+}
+```
+
+If it's not the case (`ProductAttribute` doesn't implement `FinderExcludable`), the product attribute exclusion menu list will not be visible and will not appear.
+
+It works the same way if you want to allow dynamic exclusion of product options
+
+
+
```
Comment (jeremycr) [inline README.md:147]: Maybe add the same code example as above ?

**Pull request merged**